libsolidity/interface/CompilerStack.cpp

```diff
@@ -1997,3 +1991,20 @@ experimental::Analysis const& CompilerStack::experimentalAnalysis() const
 	solAssert(!!m_experimentalAnalysis);
 	return *m_experimentalAnalysis;
 }
+
+void CompilerStack::formatUnimplementedFeatureError(UnimplementedFeatureError const& _unimplementedError)
+{
+	SourceLocation errorSourceLocation{};
+	if (
+		SourceLocation const* sourceLocation =
+		boost::get_error_info<langutil::errinfo_sourceLocation>(_unimplementedError)
+	)
+		errorSourceLocation = *sourceLocation;
+
+	std::string const* comment = _unimplementedError.comment();
+	m_errorReporter.unimplementedFeatureError(
+		1834_error,
+		errorSourceLocation,
+		(comment && !comment->empty()) ? *comment : ""
+	);
+}
```

Comment:
This could actually be shortened to a one-liner with just two more generic helpers (that would be useful also in other situations):

```c++
m_errorReporter.unimplementedFeatureError(1834_error, _error.sourceLocation(), stringOrDefault(_error.comment()));
```
- `util::Exception::sourceLocation()` - we already have `comment()` and `lineInfo()` there. We could just as well add source location. Alternatively, this could be defined in `UnimplementedFeatureError`, but I think more exception types may carry location.
- `stringOrDefault()` overload that takes `string const*`, next to the current one that takes `string`.
    - We could also set the default there to `""`.
    - The function could be moved to `CommonData.h` to make it more widely available (it's currently defined in `libsolutil/Assertions.h`).